middleware/csrf_test.go

```diff
@@ -358,3 +359,31 @@ func TestCSRFConfig_skipper(t *testing.T) {
 		})
 	}
 }
+
+func TestCSRFErrorHandling(t *testing.T) {
+	cfg := CSRFConfig{
+		ErrorHandler: func(err error, c echo.Context) error {
+			return echo.NewHTTPError(http.StatusTeapot, "error_handler_executed")
+		},
+	}
+	expectedErr := echo.NewHTTPError(http.StatusTeapot, "error_handler_executed")
+
+	e := echo.New()
+	e.POST("/", func(c echo.Context) error {
+		return c.String(http.StatusNotImplemented, "should not end up here")
+	})
+
+	e.Use(CSRFWithConfig(cfg))
+
+	req := httptest.NewRequest(http.MethodPost, "/", nil)
+	res := httptest.NewRecorder()
+	e.ServeHTTP(res, req)
+
+	var response struct {
+		Message string `json:"message"`
+	}
+
+	assert.NoError(t, json.Unmarshal(res.Body.Bytes(), &response))
+	assert.Equal(t, expectedErr.Code, res.Code)
+	assert.Equal(t, expectedErr.Message, response.Message)
```

Comment:
Seems ok but please change test not deal with unmarshaling. We usually do something like that:

```go
			assert.Equal(t, http.StatusTeapot, res.Code)
			assert.Equal(t, "{\"message\":\"error_handler_executed\"}\n", res.Body.String())
```

Reply:
Done.